File: temas/delegacion/Lavadora.java

public class Lavadora{
    private LavadoraAutomatica LavadoraAutomatica;
    
    public Lavadora() {
        this.LavadoraAutomatica = new LavadoraAutomatica();
    }
    
    public void lavarRopa() {
        LavadoraAutomatica.lavar();
    }
}

File: temas/delegacion/Lavadora.scala
implicit val lavadoraAutomaticaAdapter: LavadoraAutomaticaAdapter = new LavadoraAutomaticaAdapter(new LavadoraAutomatica())

class Lavadora(implicit lavadora: LavadoraInterface) {
  def lavarRopa(): Unit = {
    lavadora.lavar()
  }
}